src/Blocks/BlockPiston.cpp

```diff
@@ -86,28 +73,129 @@ bool cBlockPistonHandler::GetPlacementBlockTypeMeta(
 
 
 
-int cBlockPistonHandler::FirstPassthroughBlock(int a_PistonX, int a_PistonY, int a_PistonZ, NIBBLETYPE pistonmeta, cWorld * a_World)
+Vector3i cBlockPistonHandler::VectorFromMetaData(int a_PistonMeta)
```

Comment:
`static` functions cannot be `const` - they don't have an object to not modify :) Keep it `static` then, just rename and change the param type..